Add ASR with TTS Tutorial. Fix enhancer usage.

artbataev opened the pull request:
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [ASR]

# Changelog 
- Tutorial for using hybrid ASR-TTS Models (`ASRWithTTSModel`)
- fix enhancer loading in `ASRWithTTSModel`

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [x] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.


## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

GNroy commented on tutorials/asr/ASR_TTS_Tutorial.ipynb:
```diff
@@ -0,0 +1,797 @@
+   "metadata": {},
+   "source": [
+    "The tutorial demonstrated the main concepts related to hybrid ASR-TTS models to finetune ASR models and train new ones from scratch. \n",
+    "The ability to achieve good text-only adaptation results is demonstrated by finetuning a small Conformer model on text-only data from the AN4 dataset."
```
I believe the Conclusion section is optional. You can leave it though if you like it. Or you can make it "Further Reading" or something.

racoiaws replied:
I think it's nice to have this section. I'd suggest renaming this to "Summary"

artbataev replied:
Renamed to "Summary"

GNroy reviewed (commented):
A good tutorial overall.
Please see the comments below and do grammar proofreading.
I'd also recommend adding a section about the Enhancer (what is it, maybe how to build and train it).

artbataev commented on tutorials/asr/ASR_TTS_Tutorial.ipynb:
```diff
@@ -0,0 +1,797 @@
+    "\n",
+    "The architecture is shown in the figure. \n",
+    "\n",
+    "The model can take text or audio as input during training. In the case of audio input, a mel spectrogram is extracted as usual and passed to ASR neural network. In the case of textual input, the mel spectrogram generator produces a spectrogram on the fly from the text. The spectrogram is improved by the enhancer (if present) and fed into the ASR model. \n",
```
Fixed

artbataev commented on tutorials/asr/ASR_TTS_Tutorial.ipynb:
```diff
@@ -0,0 +1,797 @@
+    "Text-only manifest should contain three fields:\n",
+    "- `text`: target text for ASR model\n",
+    "- `tts_text`: text to use as a source for TTS model (unnormalized)\n",
+    "- `tts_text_normalized`: text to use as a source for TTS model (unnormalized)\n",
```
Fixed

artbataev commented on tutorials/asr/ASR_TTS_Tutorial.ipynb:
```diff
@@ -0,0 +1,797 @@
+    "* TTS Mel Spectrogram Generator (currently, only `FastPitch` model is supported)\n",
+    "* Enhancer model (optional)\n",
+    "\n",
+    "Also, the config allows to specify text-only dataset.\n",
```
Done

artbataev commented:
@GNroy, @racoiaws, I addressed your comments. 

I also added a section about TTS models before the summary. 

Also, tested on Goole Colab, everything works, but for now I should set `BRANCH=cherry-pick-main-de52d957e63a6e5cc22b2e9463420223aacca533`. This will be fixed after merging #7013 into main.

GNroy reviewed (approved):
LGTM, thanks!
